netket/operator/_abstract_operator.py

```diff
@@ -123,224 +104,3 @@ def conjugate(self, *, concrete=False) -> "AbstractOperator":
-        low = 0
-        for i in range(out.shape[0]):
-            old_out = out[i]
-            out[i] = out[i] - low
-            low = old_out
-
-        return out
-
-    def to_sparse(self) -> _csr_matrix:
-        r"""Returns the sparse matrix representation of the operator. Note that,
-        in general, the size of the matrix is exponential in the number of quantum
-        numbers, and this operation should thus only be performed for
-        low-dimensional Hilbert spaces or sufficiently sparse operators.
-
-        This method requires an indexable Hilbert space.
-
-        Returns:
-            The sparse matrix representation of the operator.
-        """
-        concrete_op = self.collect()
-        hilb = self.hilbert
-
-        x = hilb.all_states()
-
-        sections = np.empty(x.shape[0], dtype=np.int32)
-        x_prime, mels = concrete_op.get_conn_flattened(x, sections)
-
-        numbers = hilb.states_to_numbers(x_prime)
-
-        sections1 = np.empty(sections.size + 1, dtype=np.int32)
-        sections1[1:] = sections
-        sections1[0] = 0
-
-        ## eliminate duplicates from numbers
-        # rows_indices = compute_row_indices(hilb.states_to_numbers(x), sections1)
-
-        return _csr_matrix(
-            (mels, numbers, sections1),
-            shape=(self.hilbert.n_states, self.hilbert.n_states),
-        )
-
-        # return _csr_matrix(
-        #    (mels, (rows_indices, numbers)),
-        #    shape=(self.hilbert.n_states, self.hilbert.n_states),
-        # )
-
-    def to_dense(self) -> np.ndarray:
-        r"""Returns the dense matrix representation of the operator. Note that,
-        in general, the size of the matrix is exponential in the number of quantum
-        numbers, and this operation should thus only be performed for
-        low-dimensional Hilbert spaces or sufficiently sparse operators.
-
-        This method requires an indexable Hilbert space.
-
-        Returns:
-            The dense matrix representation of the operator as a Numpy array.
-        """
-        return self.to_sparse().todense().A
-
-    def apply(self, v: np.ndarray) -> np.ndarray:
-        op = self.to_linear_operator()
-        return op.dot(v)
-
-    def __call__(self, v: np.ndarray) -> np.ndarray:
-        return self.apply(v)
-
-    def conj(self, *, concrete=False) -> "AbstractOperator":
-        return self.conjugate(concrete=False)
-
-    def to_linear_operator(self):
-        return self.to_sparse()
-
-    def _get_conn_flattened_closure(self):
-        raise NotImplementedError(
-            """
-            _get_conn_flattened_closure not implemented for this operator type.
-            You were probably trying to use an operator with a sampler.
-            Please report this bug.
-
-            numba4jax won't work.
-            """
-        )
-
-    def __repr__(self):
-        return f"{type(self).__name__}(hilbert={self.hilbert})"
-
-    def __matmul__(self, other):
```

Comment:
everything below here should stay in here.
i will rewrite that stuff with dispatch at some point, but for now let's leave it here